romancal/dark_current/dark_current_step.py

```diff
@@ -133,10 +135,10 @@ def dark_output_data_as_ramp_model(out_data, input_model):
     # Removing integration dimension from variables (added for stcal
     # compatibility)
     # Roman 3D
-    out_model.data = out_data.data[0]
+    out_model.data = u.Quantity(out_data.data[0], out_model.data.unit, dtype=out_model.data.dtype)
     out_model.groupdq = out_data.groupdq[0]
     # Roman 2D
     out_model.pixeldq = out_data.pixeldq
-    out_model.err = out_data.err[0]
+    out_model.err = u.Quantity(out_data.err[0], out_model.err.unit, dtype=out_model.err.dtype)
```

Comment:
If these changes were after dark_sub.do_correction, these lines could be deleted?

Reply:
No, one is dark_data and one is out_data.